### PR TITLE
[FRONT-53] Sidebars – layout and interaction

### PR DESCRIPTION
**Pull request by mondoreale**
- [x] ~~Can we go with better permission names? (Get → Get details, etc.)~~ → [`FRONT-68`](https://linear.app/streamr/issue/FRONT-68/share-sidebar-→-better-permission-names)
- [x] Role radio buttons don't fit sometimes (for streams "Owner" is cut off)

### CODE REVIEW

File: app/src/shared/components/Sidebar/index.jsx [hunk 1]
```diff
@@ -64,8 +64,25 @@ const Sidebar = ({ className, isOpen, onClose, children }: Props) => {
 
 export { Select }
 
+const Container = styled.div`
+    padding: 24px 32px;
```
Comment (mondoreale) [inline app/src/shared/components/Sidebar/index.jsx:68]: `Sidebar.Container` (in this file "just" `Container`) defines spacing around all key sidebar components. It's used all over the place. Reasonable DRY FTW!

File: app/src/shared/components/Tooltip/index.jsx [hunk 2]
```diff
@@ -67,4 +67,8 @@ Tooltip.defaultProps = {
     placement: Tooltip.TOP,
 }
 
+Object.assign(Tooltip, {
+    Root,
```
Comment (mondoreale) [inline app/src/shared/components/Tooltip/index.jsx:71]: Promote `Root` to by stylable outside this file. 💪

File: app/src/userpages/components/ShareSidebar/Footer.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+const Footer = styled(UnstyledFooter)`
+    border-top: 1px solid #efefef;
+    display: flex;
+
+    ${Button} {
+        min-width: 96px;
+    }
+
+    > div:first-child {
+        flex-grow: 1;
```
Comment (mondoreale) [inline app/src/userpages/components/ShareSidebar/Footer.jsx:52]: IIRC, previously it was `margin-right: auto` but I believe that if you flex you gotta flex hard, thus `flex-grow: 1`. 😅

File: app/src/userpages/components/ShareSidebar/UserList.jsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                        resourceType={resourceType}
+                        updatePermission={updatePermission}
+                        userId={userId}
+                        userPermissions={userPermissions}
+                    />
+                    {!!userErrors[userId] && (
+                        <Sidebar.Container as={ErrorMessage}>
+                            {userErrors[userId].message}
+                        </Sidebar.Container>
+                    )}
```
Comment (mondoreale) [inline app/src/userpages/components/ShareSidebar/UserList.jsx:51]: Here's something new. `UserPermissions` became `Sidebar.Container` thus they come with padding. Rendering errors within that structure wouldn't end up well (noticable padding around a red error box). I render errors outside (after) `UserPermissions` then. Btw, `ErrorMessage` is also a `Sidebar.Container` here!

File: app/src/userpages/i18n/en.po [hunk 5]
```diff
@@ -1125,7 +1125,7 @@ msgid "modal.shareResource.linkCopied"
 msgstr "Link copied"
 
 msgid "modal.shareResource.warnUnsavedChanges"
-msgstr "To update your permissions please save your changes. To discard them, click cancel."
+msgstr "To update your permissions please **save** your changes. To discard them, click **cancel**."
```
Comment (mondoreale) [inline app/src/userpages/i18n/en.po:1128]: We can render markdown stuff now. Example:
```jsx
<Md inline>
    {I18n.t('modal.shareResource.warnUnsavedChanges')}
</Md>
```
gives
```html
To update your permissions please <strong>save</strong> your
changes. To discard them, click <strong>cancel</strong>.
```
`inline` flag makes it skip `<p>` around the output cause only certain "inline" tags are allowed in that scenario, FYI.

**Review by mondoreale — COMMENTED** [review]
Left a little self-review for you guys. This PR is sizable but not massive. Hope you can read right through it w/o questions. LMK otherwise. Cheers!

**Comment by mondoreale** [comment]
@mattatgit I intentionally left a couple paddings and margins different than in the design. They work well in the code (no edge cases). See the spec below:

Spec

Open to modify them. LMK.

**Comment by mondoreale** [comment]
Updated todos (see description), FYI.

**Comment by mattatgit** [comment]
@mondoreale Looks pretty good to me sir.

**Comment by mondoreale** [comment]
Guys, I pushed a little something for the radio buttons. They all fit now. Created a separate ticket for the permission names (see description).

We're all set.

**Comment by juhah** [comment]
Code & styling changes look good 👍 Here's a couple UI notes:

Clicking outside the checkboxes closes the permission group where before only clicking the title closed it:
ezgif-2-09f93ba589e3

Should the **X** icon act as cancel, now it's the same as clicking outside:
ezgif-2-2d94be37f84f

**Review by tumppi — APPROVED** [review]
I noticed that on medium breakpoint the `Copy link` button is not aligned with other buttons:

Screenshot 2020-09-17 at 13 36 05

Otherwise looks good to me!

**Comment by mondoreale** [comment]
All good points guys! Addressing them right this second. 🔥

**Comment by mondoreale** [comment]
@mattatgit should the "x" button act as a "cancel" button? We'd have to dig a little bit deeper into the `Sidebar` component internals to make that change. Not a lot. It's not a no-brainer tho.

cc @juhah

**Comment by mattatgit** [comment]
@mondoreale @juhah No, it should just be disabled while in the state of waiting for user decision. They can either cancel or save.

**Comment by mondoreale** [comment]
Good! Thx

**Review by timoxley — COMMENTED** [review]
good changes

**Pull request merged**